cs-minibot-platform-src/src/main/java/examples/gobot/GoBot.java

```diff
@@ -0,0 +1,375 @@
+                                System.out.println(navigator.destination);
+                                index = (index + 1) % max;
+                            }
+                        }
+                        if (course.getStartArea().contains(vc.x, vc.y)) {
+                            if (!crossedLapLine && !reachedMiddle) {
+                                this.crossedLapLine = true;
+                            } else if (crossedLapLine && reachedMiddle) {
+                                this.lapsDone++;
+                                this.reachedMiddle = false;
+                                this.lapTimes.add(System.nanoTime());
+                                System.out.println(lapTimes());
+                            }
+                        } else if (course.getMiddleArea().contains(vc.x, vc.y)) {
+                            if (!reachedMiddle) {
+                                this.reachedMiddle = true;
+                            }
+                        }
+                    }
+                }
+                else{
+                    navigator.run();
+                }
+
+            }
+            else {
+                return;
+            }
+            try {
+                Thread.sleep(1);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    private class Navigator extends Thread {
+        private boolean destinationReached;
+        private VisionCoordinate destination;
+
+        private static final double DISTANCE_THRESHOLD = 0.08;
+        private static final double ANGLE_THRESHOLD = Math.PI/(10);
+        private static final int MAX_SPEED = 100;
+        private static final int MIN_SPEED = 50;
+
+        @Override
+        public void run() {
+            while (true) {
+                calcRoute();
+                try {
+                    Thread.sleep(20);
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+
+        public boolean destinationReached() {
+            return destination == null || destinationReached;
+        }
+
+        public void goToDestination(VisionCoordinate vc) {
+            this.destination = vc;
+            destinationReached = false;
+        }
+
+        private double mod(double a, double n){
+            return a - Math.floor(a/n)*n;
+        }
+
+        public void calcRoute(){
+            VisionCoordinate vc;
+            List<VisionObject> locs = BaseStation.getInstance()
+                    .getVisionManager
+                            ().getAllLocationData();
+
+            if (locs.size() == 0 || locs.get(0) == null){
+                vc = null;
+                fwm.setWheelPower(0,0,0,0);
+                return;
+            }
+            else {
+                vc = locs.get(0).coord;
+            }
+
+            if (aiType == 0){
```

Comment:
it would be ideal for both route calculators to be in their own classes that implement a route calculating interface.. but i guess you don't have to do that if you don't have time.